azurerm/resource_arm_data_factory_linked_service_sql_server.go

```diff
@@ -0,0 +1,357 @@
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("data_factory_name", dataFactoryName)
+
+	sqlServer, ok := resp.Properties.AsSQLServerLinkedService()
+	if !ok {
+		return fmt.Errorf("Error classifiying Data Factory Linked Service SQL Server %q (Data Factory %q / Resource Group %q): Expected: %q Received: %q", name, dataFactoryName, resourceGroup, datafactory.TypeSQLServer, *resp.Type)
+	}
+
+	d.Set("additional_properties", sqlServer.AdditionalProperties)
+
+	if sqlServer.Description != nil {
+		d.Set("description", *sqlServer.Description)
+	}
+
+	annotations := flattenDataFactoryLinkedServiceAnnotations(sqlServer.Annotations)
+	if err := d.Set("annotations", annotations); err != nil {
+		return fmt.Errorf("Error setting `annotations`: %+v", err)
+	}
+
+	parameters := flattenDataFactoryLinkedServiceParameters(sqlServer.Parameters)
+	if err := d.Set("parameters", parameters); err != nil {
+		return fmt.Errorf("Error setting `parameters`: %+v", err)
+	}
+
+	if connectVia := sqlServer.ConnectVia; connectVia != nil {
+		if connectVia.ReferenceName != nil {
+			d.Set("integration_runtime_name", *connectVia.ReferenceName)
+		}
+	}
+
+	if properties := sqlServer.SQLServerLinkedServiceTypeProperties; properties != nil {
+		if properties.ConnectionString != nil {
+			val, ok := properties.ConnectionString.(string)
+			if !ok {
+				log.Printf("[DEBUG] Skipping connection string %q since it's not a string", val)
+			} else {
+				d.Set("connection_string", properties.ConnectionString.(string))
+			}
+		}
+	}
+
+	return nil
+}
+
+func resourceArmDataFactoryLinkedServiceSQLServerDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataFactoryLinkedServiceClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	dataFactoryName := id.Path["factories"]
+	name := id.Path["linkedservices"]
+
+	response, err := client.Delete(ctx, resourceGroup, dataFactoryName, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(response) {
+			return fmt.Errorf("Error deleting Data Factory Linked Service SQL Server %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+		}
+	}
+
+	return nil
+}
+
+// Because the password isn't returned from the api in the connection string, we'll check all
+// but the password string and return true if they match.
+func azureRmDataFactoryLinkedServiceConnectionStringDiff(k, old string, new string, d *schema.ResourceData) bool {
+	oldSplit := strings.Split(strings.ToLower(old), ";")
+	newSplit := strings.Split(strings.ToLower(new), ";")
+
+	sort.Strings(oldSplit)
+	sort.Strings(newSplit)
+
+	// We need to remove the password from the new string since it isn't returned from the api
+	for i, v := range newSplit {
+		if strings.HasPrefix(v, "password") {
```

Comment:
IIRC the keys for a connection string can be case insensitive; as such we may want to make this:

```suggestion
		if strings.HasPrefix(strings.ToLower(v), "password") {
```

Reply:
This was an issue that I took care of a few lines up with:
```
oldSplit := strings.Split(strings.ToLower(old), ";")
newSplit := strings.Split(strings.ToLower(new), ";")
```